### PR TITLE
Add puppet-agent 1.2.7 release notes.

### PR DESCRIPTION
**Pull request by gguillotte**
Adds release notes.

### CODE REVIEW

**Comment by gguillotte** [comment]
@joshcooper @kylog @nfagerlund

File: source/puppet-agent/_release_notes.md [hunk 1]
```diff
@@ -1,10 +1,13 @@
 ### Release Notes
 
+* **1.2.7**, Oct. 29, 2015: Adds or improves support for AIX 5.3, [6.1](https://tickets.puppetlabs.com/browse/PA-32), and [7.1](https://tickets.puppetlabs.com/browse/PA-33) on PowerPC, Solaris 11 on [SPARC](https://tickets.puppetlabs.com/browse/PA-4) and [i386](https://tickets.puppetlabs.com/browse/PA-5), and Enterprise Linux 4. Updates Ruby, [Facter](/facter/3.1/release_notes.html#facter-311), and [Hiera](/hiera/latest/release_notes.html#hiera-304).
```
Comment (joshcooper) [inline source/puppet-agent/_release_notes.md]: Could you link AIX 5.3 to https://tickets.puppetlabs.com/browse/PA-41 (just moved the RE ticket to PA like the others)

**Comment by joshcooper** [comment]
@gguillotte I just remembered that the new platform additions (EL4, AIX, Solaris 11) are PE only. This means we will not be providing open source packages for those platforms when we ship puppet-agent 1.2.7, but they will be available for download for PE customers when PE 2015.2.3 comes out in Nov.

Since we're not shipping any bits for those platforms at this time, I'm thinking we should remove references to those platforms, and handle it in PE documentation. Thoughts? /cc @ipeldridge @nfagerlund

**Comment by ipeldridge** [comment]
@joshcooper I've already added them to the docs for PE 2015.2.3. Thanks.

**Comment by kylog** [comment]
I agree with removing Sparc/AIX/EL4 from the 1.2.7 release notes.

**Comment by gguillotte** [comment]
Removed. Does this affect the Facter and Hiera release notes as well?

**Comment by kylog** [comment]
For any of the components (e.g. puppet/facter/hiera), I think it's correct/appropriate to mention fixes in the source for any/all platforms, including the packaged-for-PE-only platforms. So my comment above was specific to puppet-agent. (Otoh, once we open-source puppet-agent, we might want to fine-tune this reasoning.)

Re hiera specifically, though, IIRC there are no functional changes, so the hiera release notes would just be like for 3.0.3, no functional change: https://docs.puppetlabs.com/hiera/3.0/release_notes.html.

I know this may be kind of hair-splitting. Do those approaches make sense?

**Comment by gguillotte** [comment]
Yep! Thanks @kylog .

**Comment by joshcooper** [comment]
:+1: to what @kylog said though there was one (maint) commit to hiera that changes an error message when failing to retrieve hiera data. Not sure if we want to mention that for 3.0.4 specifically?

**Comment by joshcooper** [comment]
Here are top-level queries:

introduced in puppet-agent 1.2.7: https://tickets.puppetlabs.com/issues/?filter=15776
fixed in puppet-agent 1.2.7: https://tickets.puppetlabs.com/issues/?filter=15775

**Comment by gguillotte** [comment]
Amended to clarify that PE packages will include AIX and Solaris 11.

File: source/puppet-agent/_release_notes.md [hunk 2]
```diff
@@ -1,10 +1,13 @@
 ### Release Notes
 
+* **1.2.7**, Oct. 29, 2015: Updates Ruby, [Facter](/facter/3.1/release_notes.html#facter-311), and [Hiera](/hiera/latest/release_notes.html#hiera-304). This version also [moves the OS X bill of materials](https://tickets.puppetlabs.com/browse/PA-21) to `/usr/local/share` on all supported OS X platforms to ensure future compatibility with OS X 10.11 (El Capitan), and [adds a root certificate](https://tickets.puppetlabs.com/browse/PA-20) in Enterprise Linux 4 to enable Forge access.
```
Comment (joshcooper) [inline source/puppet-agent/_release_notes.md]: I think we only bumped Ruby to 1.0.2d for Windows, as the other platforms were already there.

Also could you add a note that puppet was updated too?

**Comment by kayayarai** [comment]
:+1:

**Comment by joshcooper** [comment]
@gguillotte looks like my comment is getting lost in the force push. Could you add a note that Puppet was updated too (right now it just mentions Facter and Hiera). Also we only updated Ruby to 1.0.2d for Windows, but not for any other platform.

**Comment by gguillotte** [comment]
@joshcooper Bumped.

**Comment by joshcooper** [comment]
:+1:

**Pull request merged**